src/charm.py

```diff
@@ -416,8 +428,14 @@ def _are_db_relations_ready(self) -> bool:
         if not self._stored.redis_relation:  # type: ignore
             self.model.unit.status = WaitingStatus("Waiting for redis relation")
             return False
-        if self._get_redis_relation_data()[0] == "":
-            self.model.unit.status = WaitingStatus("Waiting for redis relation to initialize")
+        try:
+            if (
+                self._get_redis_relation_data()[0] in ("", "None")
+                or self._get_redis_relation_data()[1] == 0
+            ):
+                self.model.unit.status = WaitingStatus("Waiting for redis relation to initialize")
+                return False
+        except MissingRedisRelationDataError:
             return False
```

Comment:
I think we'd want to set some kind of status on the unit here wouldn't we?

Reply:
Fixed